python_modules/dagster/dagster/_core/storage/schedules/sql_schedule_storage.py

```diff
@@ -446,6 +455,55 @@ def purge_ticks(
         with self.connect() as conn:
             conn.execute(query)
 
+    def add_auto_materialize_asset_evaluations(
+        self,
+        evaluation_id: int,
+        asset_evaluations: Sequence[AutoMaterializeAssetEvaluation],
+    ):
+        check.invariant(len(asset_evaluations) > 0)
+        with self.connect() as conn:
+            bulk_insert = AssetPolicyEvaluationsTable.insert().values(
+                [
+                    {
+                        "evaluation_id": evaluation_id,
+                        "asset_key": decision.asset_key.to_string(),
+                        "asset_evaluation_body": serialize_value(decision),
+                        "num_requested": decision.num_requested,
+                    }
+                    for decision in asset_evaluations
+                ]
+            )
+            conn.execute(bulk_insert)
+
+    def get_auto_materialize_asset_evaluations(
```

Comment:
At some point in the nearish future, we're probably going to want to support a view that aggregates evaluations across multiple assets. Would it make sense to plan ahead for that by accepting multiple asset keys here?

Reply:
Offline discussion:
For the aggregate view, we wouldn't want to deserialize the evaluation bodies for each included asset. In a followup, we'll add a separate method that does a groupby evaluation id and sums num_requested, num_skipped, and num_discarded